client/state/comments/reducer.js

```diff
@@ -0,0 +1,266 @@
+			if ( commentNodeChanged ) {
+				commentsTree.update( 'fetchedCommentsCount', ( fetchedCommentsCount ) => fetchedCommentsCount + 1 );
+
+				if ( commentsTree.getIn( [ comment.ID, 'parentId' ] ) === null ) {
+					commentsTree.update( 'children', ( children ) => children.push( comment.ID ) );
+				}
+			}
+		} );// forEach comments
+	} ); // withMutation
+
+	return newTree;
+}
+
+/***
+ * Removes comment from the comment tree
+ * @param {Immutable.Map} oldTree Immutable map comments tree instance of the shape Map<id, CommentNode>{ children: List<id> }
+ * @param {Number} commentId the requested comment id for removal
+ * @returns {Immutable.Map} Immutable map comments tree instance of the shape Map<id, CommentNode>{ children: List<id> }
+ */
+function removeComment( oldTree, commentId ) {
+	const newTree = oldTree.withMutations( ( commentsTree ) => {
+		const commentNode = commentsTree.get( commentId );
+
+		if ( commentNode ) {
+			if ( commentNode.get( 'parentId' ) ) {
+				commentsTree.updateIn( [ commentNode.get( 'parentId' ), 'children' ], ( children ) => children.filter( ( childId ) => childId !== commentId ) );
+			}
+
+			commentsTree.delete( commentId );
+			commentsTree.update( 'fetchedCommentsCount', ( fetchedCommentsCount ) => fetchedCommentsCount - 1 );
+		}
+	} );
+
+	return newTree;
+}
+
+/***
+ * Stores comments tree per commentTargetId (siteId, postId)
+ * @param {Immutable.Map} state redux state comments tree
+ * @param {Object} action redux action
+ * @returns {Object} redux state
+ */
+export function tree( state, action ) {
+	switch ( action.type ) {
+		case COMMENTS_RECEIVE:
+			return buildCommentsTree( state, action.comments );
+
+		case COMMENTS_REMOVE_COMMENT:
+			return removeComment( state, action.commentId );
+
+		case DESERIALIZE:
+			// the combined reducer already handled fromJS() for us, the only thing left
+			// is fix the numeric keys, which JS has as strings.
+			return state.mapKeys( ( k ) => ( ! isNaN( k ) ) ? parseInt( k, 10 ) : k );
+			break;
+
+		case SERIALIZE:
+			// no need to do anything here, since the combined reducer will handle it for us.
+			break;
+	}
+
+	return state;
+}
+
+/***
+ * Stores information regarding requests status per requestId
+ * @param {Object} state redux state
+ * @param {Object} action redux action
+ * @returns {Object} redux state
+ */
+export function queries( state = Immutable.Map(), action ) {
+	switch ( action.type ) {
+		case COMMENTS_REQUEST:
+		case COMMENTS_REQUEST_SUCCESS:
+		case COMMENTS_REQUEST_FAILURE:
+			return state.set( action.requestId, action.type );
+	}
+
+	return state;
+}
+
+/***
+ * Stores information regarding the latest (most recent) date we saw
+ * on a comment for a given commentTargetId
+ * @param {Date} state redux state, current latest comment date
+ * @param {Object} action redux action
+ * @returns {Object} redux state
+ */
+export function latestCommentDate( state, action ) {
```

Comment:
Using `undefined` works, but it unfortunately goes against Redux convention. I know you're using a different combine function, but it'd be great to follow convention when we can. See the notes section in: [combineReducers](http://redux.js.org/docs/api/combineReducers.html)

> If the state given to it is undefined, it must return the initial state for this specific reducer. According to the previous rule, the initial state must not be undefined either.